### PR TITLE
Background Plot: Handle errors when loading from save file

### PR DESCRIPTION
**Pull request by mihai-stfc**
### Instrument(s)

Technique, Muon

### Story/Acceptance criteria

The background plot task does not fall over if it's save files are corrupted

### Description of work

Handle invalid data point when appending to dataset, set `run_numbers_match` to False on invalid `saved_run_number`/`current_run_number`, handle row index & datetime format errors.
Display corrupt file warnings and attempt to load and plot using the valid data from the file.

### Issue/Ticket Reference

https://github.com/ISISComputingGroup/IBEX/issues/5582

### Tests

Manual tests. 

### To test

[Background Script IOC](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Background-Script-IOC)

Configuration, `background_script.py`, `init_<name>.py`, IOC logs and "corrupted" save file used for testing:
[Ticket5582.zip](https://github.com/ISISNeutronMuon/InstrumentScripts/files/5822006/Ticket5582.zip)

Run the background plot as normal via the `BGRSCRPT_01` IOC, check that the plot is working as intended.
Edit the `background_plot_data_01.csv` file to corrupt the data.
Restart IOC, make sure the dataset matches (`self.saved_data_matches_current_dataset()`) in `BackgroundPlot.start` so the script loads the save file instead of creating a new file.
Check that the IOC does not get stuck in a restart loop due to failing to load the save file.

---

#### Code Review

- [ ] Is the story/acceptance criteria fulfilled?
- [ ] Is the code of an acceptable quality?
- [ ] Are the tests sufficient?
- [ ] Do the changes function as described and is it robust?
- [ ] Are the changes able to work across all intended instruments?

### Final Steps
- [ ] Are there any changes to instrument configurations required?
- [ ] Are there any changes to instrument scripts required, e.g. change on script signature, default argument and have these been communicated?
- [ ] Does the script need to be deployed onto the instrument?

### CODE REVIEW

**Review by RaiBishal — CHANGES_REQUESTED** [review]
Looks good! however, we should start adding tests. I understand there are no tests currently for this module but we can start by adding couple for `load_data_from_save`.

File: technique/muon/tests/test_background_plot.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    def setUp(self):
+        self.background_plot = BackgroundPlot(interval=None, ioc_number=1)
+
+    def tearDown(self):
+        pass
+
+    @patch('builtins.print')
+    def test_GIVEN_corrupted_file_WHEN_load_data_THEN_display_warnings_and_no_exception(self, m_print):
+        pass
```
Comment (Tom-Willemsen) [inline technique/muon/tests/test_background_plot.py:29]: remove `pass`

File: technique/muon/tests/test_background_plot.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+THIS_DIR = os.path.dirname(os.path.abspath(__file__))
+
+
+class TestLoadDataFromSave(unittest.TestCase):
+
+    def setUp(self):
+        self.background_plot = BackgroundPlot(interval=None, ioc_number=1)
+
+    def tearDown(self):
+        pass
```
Comment (Tom-Willemsen) [inline technique/muon/tests/test_background_plot.py:25]: remove this method if it doesn't do anything

File: technique/muon/tests/data/corrupted_plot_data_01.csv [hunk 3]
```diff
@@ -0,0 +1,602 @@
+# run_number:00000039
```
Comment (Tom-Willemsen) [inline technique/muon/tests/data/corrupted_plot_data_01.csv:1]: Usually we shouldn't have large data files for a unit test.

Can you remove this and split the unit test you've added into separate tests for the different cases:
- Date format wrong
- No data present
- Wrong number of fields
- etc

**Review by Tom-Willemsen — CHANGES_REQUESTED** [review]

**Review by Tom-Willemsen — APPROVED** [review]

**Pull request merged**